telegram_handler.rb

```diff
@@ -115,7 +128,49 @@ def self.handle_input(bot, message)
         #   bot.api.send_message(chat_id: message.chat.id, text: "Number of months tracked: #{number_of_months.round(1)}")
         #   bot.api.send_message(chat_id: message.chat.id, text: "Averaging #{average_number_of_moods.round(1)} per day")
         when "/start"
+          kb=$kb;
+          Mood::Database.database[:moodlabels].where(:chat_id => message.chat.id).each do |n|
+            _mood = n[:mood].to_i
+            _label = n[:label].to_s
+            kb[5 - _mood] = Telegram::Bot::Types::InlineKeyboardButton.new(text:"#{_mood}: #{_label}", callback_data: _mood)
+          end          
+          answers = Telegram::Bot::Types::InlineKeyboardMarkup.new(inline_keyboard: kb)
+
           bot.api.send_message(chat_id: message.chat.id, reply_markup: answers, text: "🙋‍♂️ Welcome to feelike! 🙋‍♀️\nI will help you keep track of your mood.\nThree times a day I will ask you how do you feel at the moment.\nYou can use my special moods keyboard or just type in a 0-5 number (5 being the happiest).\nWhen you want to see your progress just send me '/graph'\n🦋\nSo let's give it a try! how do you feel like right now?")
+        when "/mood"  
+          kb=$kb;
+          Mood::Database.database[:moodlabels].where(:chat_id => message.chat.id).each do |n|
+            _mood = n[:mood].to_i
+            _label = n[:label].to_s
+            kb[5 - _mood] = Telegram::Bot::Types::InlineKeyboardButton.new(text:"#{_mood}: #{_label}", callback_data: _mood)
+          end          
+          answers = Telegram::Bot::Types::InlineKeyboardMarkup.new(inline_keyboard: kb)
+          bot.api.send_message(chat_id: message.chat.id, reply_markup: answers, text: "How do you feel like? Share your mood")
+        when "/setlabel"
+          bot.api.send_message(chat_id: message.chat.id, text: "To set a mood's label use format: '/setlabel # Mood label'")
+        when /\/setlabel\ /
+          label_content = message.text.split("/setlabel ").last
+          label_mood = label_content[0]
+          label_text = label_content[2,label_content.length - 1]
+          
+          if (label_mood=="0" or label_mood=="1" or label_mood=="2" or label_mood=="3" or label_mood=="4" or label_mood=="5")
+            Mood::Database.database[:moodlabels].replace({
+              chat_id: message.chat.id,
+              mood: label_mood,
+              label: label_text
+            })
```

Comment:
Else return an error message?

Reply:
There is an else there. Look two lines down